<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="GEObridge">
<title>GEObridge -- an R package to connect the GEO database and researcher • GEObridge</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GEObridge -- an R package to connect the GEO database and researcher">
<meta property="og:description" content="GEObridge">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GEObridge</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignettes.html">GEObridge -- an R package to connect the GEO database and researcher</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/yuefang/GEObridge/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>GEObridge -- an R package to connect the GEO database and researcher</h1>
                        <h4 data-toc-skip class="author">Yue Fang <a href="mailto:fangyue@iu.edu" class="email">fangyue@iu.edu</a>
</h4>
            
            <h4 data-toc-skip class="date">Dec. 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/yuefang/GEObridge/blob/HEAD/vignettes/vignettes.Rmd" class="external-link"><code>vignettes/vignettes.Rmd</code></a></small>
      <div class="d-none name"><code>vignettes.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Data Science (DS) becoming more and more important recently as everyone already feels it. Data analysis has infiltrated every industrial and academic field including news, sport, biopharma, big tech, and precision medicine. Specifically, precision medicine looks at the genetics, disease environment, and diet of a person in order to select the treatment that could work best for the patient, which is significantly dependent on large-scale sequencing data. The National Center for Biotechnology Information (NCBI) released a public repository, named Gene Expression Omnibus (GEO) [2-4], for high-throughput molecular abundance experiment data, providing free distribution and shared access to comprehensive datasets. Till 10/26/2022, GEO contains 216,550 datasets and over 4,446,632 samples covering Homo sapiens, Mus musculus, and all other organisms.</p>
<p>As the number of datasets accumulated in GEO, downloading and querying data from the GEO database became a difficult problem due to its large scale and category complexity of it. A lot of methods and tools were developed during the last two decades to solve this urgent issue and satisfy the unmet needs of different user groups including bioinformaticians, doctors without coding experience, biologists who focus on cancer immunotherapy development, and so on. GEOquery is one popular and outstanding R package which enable the user to download the sequencing data from the GEO database with the access ID and other references [1]. The package is available on the Bioconductor platform as well as the GEOmetadb [5]. GEO data browser [6] is the official tool of the GEO database which is similar to an internal search engine inside the GEO database. However, manual searching through thousands of datasets and millions of samples is not enough for a heavy labor research topic. Thus, GEO metadb becomes a useful tool for mining the data of GEO.</p>
<p>GEOmetadb is one of the most popular R packages among the above-mentioned R packages due to its powerful search function and clear data storage structure based on the Structured Query Language (SQL). However, there is no existing tool that enables the analysis of the downloaded table using the GEOmetadb R package. Specifically, there is no existing R function to further search and analyze the queried table and summary the category based on the interested platform or organism in the GEO database. To do so, we proposed to develop a novel R package named GEObridge to solve these pain points for users whose work will benefit from this R package.</p>
<p><strong>challenge 1</strong> <em>Too Many Categories Popped Out When Searching with Keyword</em></p>
<p>One advantage of the GEO database is its complete information and the biggest database in the public domain. On the other hand, this also brings up the issue which is too many categories were returned when using the searching function. This will let the user feel overwhelmed and do not know what to do with such a scale of the dataset. One solution to the above issue is to provide a filter function or parameter. With the help of the filter, the user can filter out the non-interested information in complex categories. For example, the searching function can return all the platforms (e.g. microarray, bulk RNA-seq, etc.) but the user only cares about the top 5 popular platforms. The user can set the filter parameter to 5 to remove the less frequency platform returned in the search result.</p>
<p><strong>challenge 2</strong> <em>Long Waiting Time for Search Function</em></p>
<p>Another challenge we faced during the package development is the search process is very slow. After researching a little bit about this issue, it turns out the current SQL-based query method is already a very efficient way to extract information from a large table. Meanwhile, we also learned some knowledge about the time complexity and memory complexity in the computer science major since their courses mentioned related terminology. To reduce time complexity, a common way in computer science is to increase memory complexity. Thus, our solution to the above issue is to prepare the result of the most frequently used function in advance and saved it in the package.</p>
<p><strong>challenge 3</strong> <em>No user-friendly visualization method public available</em></p>
<p>Even the above two points can be solved with our package, the user still want to find a way to visualize the result which can be used in their publication. It also help user get better understanding of the GEO dataset and meta information. To do so, we make two functions solve this challange. The first plot function is to draw barplot for the most frequent platform summaried in the download database. The user can select number of most frequent parameter such as top 5 or top 10 platforms. The second function is to draw barplot in an interactive way. The user can select interested platforms and draw a barplot dynamically thanks to the power of RShiny gadget. The user can use mice to click and select several platforms which will show at the right panel instantly.</p>
<p>This is the vignette of the <strong>GEObridge</strong> R package. In this vignette, we will use four examples to demostrate the utility and feasibility of the proposed R package. The proposed GEObridge package will use several functions from GEOmetadb package to query the full meta information from the GEO database. Then, we will illustrate the important of the proposed R package by showing the below four examples.</p>
</div>
<div class="section level2">
<h2 id="primary-aim">2. Primary Aim<a class="anchor" aria-label="anchor" href="#primary-aim"></a>
</h2>
<div class="section level3">
<h3 id="goal-1">2.1 Goal 1<a class="anchor" aria-label="anchor" href="#goal-1"></a>
</h3>
<div class="section level6">
<h6 id="user-friendly-searching-function-for-digging-more-detailed-information">User-friendly Searching Function for Digging More Detailed Information<a class="anchor" aria-label="anchor" href="#user-friendly-searching-function-for-digging-more-detailed-information"></a>
</h6>
<p>The primary goal of the GEObridge package is to build an R package that depends on the GEOmetadb package and provides more user-friendly functions to dig the information in the downloaded table. GEOmetadb can help the user download the whole database into the local laptop. However, how to efficiently use this download database is a hard problem. The first goal of the GEObridge is to provide some function that let the user search for target samples, platforms, or experiments in the downloaded SQLite file. We demonstrated the search function in the next section Code Example. The appendix also provides the demo’s full script and R environment based on <em>sessionInfo()</em> function.</p>
</div>
</div>
<div class="section level3">
<h3 id="goal-2">2.2 Goal 2<a class="anchor" aria-label="anchor" href="#goal-2"></a>
</h3>
<div class="section level6">
<h6 id="visualization-function-to-summarize-the-meta-information-in-the-geo-database">Visualization Function to Summarize the Meta Information in the GEO database<a class="anchor" aria-label="anchor" href="#visualization-function-to-summarize-the-meta-information-in-the-geo-database"></a>
</h6>
<p>The second goal of the GEObrige package is to provide a series of functions to visualize the meta information in the GEO database. The GEO database is very large and contains almost every type of data (e.g. Microarray, RNA-seq, single RNA-seq (scRNA-seq), Homo sapiens, Mus musculus). How to extract useful or interesting datasets become an important issue. Once the SQLite file was downloaded to the local laptop, we provide several functions to generate plots to show the statistics of the data. We demonstrated the visualization functions in the next section of Code Example.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="preliminary">3. Preliminary<a class="anchor" aria-label="anchor" href="#preliminary"></a>
</h2>
<p>The proposed GEObridge is to design and provide the down stream functions for the GEOmetadb package. Firstly, we will show how to install GEOmetadb package. Then, we will show the seven table which contains the most meta information we need thereafter.</p>
<p>Note, GEOmetadb pacakge is not publised on the CRAN. Thus, we cannot use <em>install.package()</em> function to install this package. Instead, we need to install the package from Bioconductor, which is a platform of open source software for bioinformatics.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"yuefang1991/GEObridge"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="check-all-meta-data-in-downloaded-sqlite-file">3.1 Check All Meta Data in Downloaded SQLite File<a class="anchor" aria-label="anchor" href="#check-all-meta-data-in-downloaded-sqlite-file"></a>
</h3>
<p>We already know the downloaded SQLite file is huge due to it contains all the meta data of GEO and stored it as high efficient tables. We will print all different tables in SQLite. In total, there are 11 tables stored in this file. Each table contains different meta information. For example, table <strong>gse</strong> contains all the GES ID for each datase and related meta information.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yuefang.github.io/GEObridge/" class="external-link">GEObridge</a></span><span class="op">)</span> <span class="co">#load our own package</span></span></code></pre></div>
<pre><code><span><span class="co">## Setting options('download.file.method.GEOquery'='auto')</span></span></code></pre>
<pre><code><span><span class="co">## Setting options('GEOquery.inmemory.gpl'=FALSE)</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">connection</span> <span class="op">=</span> <span class="fu"><a href="../reference/prepare_table.html">prepare_table</a></span><span class="op">(</span>printAllTable<span class="op">=</span><span class="cn">TRUE</span>, subTable<span class="op">=</span><span class="st">'gse'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "-----ALl tables stored in the database-----"</span></span>
<span><span class="co">##  [1] "gds"               "gds_subset"        "geoConvert"       </span></span>
<span><span class="co">##  [4] "geodb_column_desc" "gpl"               "gse"              </span></span>
<span><span class="co">##  [7] "gse_gpl"           "gse_gsm"           "gsm"              </span></span>
<span><span class="co">## [10] "metaInfo"          "sMatrix"          </span></span>
<span><span class="co">## [1] "------Print columns for your inputted sub table-----"</span></span>
<span><span class="co">##  [1] "ID"                   "title"                "gse"                 </span></span>
<span><span class="co">##  [4] "status"               "submission_date"      "last_update_date"    </span></span>
<span><span class="co">##  [7] "pubmed_id"            "summary"              "type"                </span></span>
<span><span class="co">## [10] "contributor"          "web_link"             "overall_design"      </span></span>
<span><span class="co">## [13] "repeats"              "repeats_sample_list"  "variable"            </span></span>
<span><span class="co">## [16] "variable_description" "contact"              "supplementary_file"</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">connection_tmp</span> <span class="op">=</span> <span class="fu"><a href="../reference/prepare_table.html">prepare_table</a></span><span class="op">(</span>printAllTable<span class="op">=</span><span class="cn">TRUE</span>, subTable<span class="op">=</span><span class="st">'gsm'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "-----ALl tables stored in the database-----"</span></span>
<span><span class="co">##  [1] "gds"               "gds_subset"        "geoConvert"       </span></span>
<span><span class="co">##  [4] "geodb_column_desc" "gpl"               "gse"              </span></span>
<span><span class="co">##  [7] "gse_gpl"           "gse_gsm"           "gsm"              </span></span>
<span><span class="co">## [10] "metaInfo"          "sMatrix"          </span></span>
<span><span class="co">## [1] "------Print columns for your inputted sub table-----"</span></span>
<span><span class="co">##  [1] "ID"                     "title"                  "gsm"                   </span></span>
<span><span class="co">##  [4] "series_id"              "gpl"                    "status"                </span></span>
<span><span class="co">##  [7] "submission_date"        "last_update_date"       "type"                  </span></span>
<span><span class="co">## [10] "source_name_ch1"        "organism_ch1"           "characteristics_ch1"   </span></span>
<span><span class="co">## [13] "molecule_ch1"           "label_ch1"              "treatment_protocol_ch1"</span></span>
<span><span class="co">## [16] "extract_protocol_ch1"   "label_protocol_ch1"     "source_name_ch2"       </span></span>
<span><span class="co">## [19] "organism_ch2"           "characteristics_ch2"    "molecule_ch2"          </span></span>
<span><span class="co">## [22] "label_ch2"              "treatment_protocol_ch2" "extract_protocol_ch2"  </span></span>
<span><span class="co">## [25] "label_protocol_ch2"     "hyb_protocol"           "description"           </span></span>
<span><span class="co">## [28] "data_processing"        "contact"                "supplementary_file"    </span></span>
<span><span class="co">## [31] "data_row_count"         "channel_count"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="code-example">4. Code Example<a class="anchor" aria-label="anchor" href="#code-example"></a>
</h2>
<p>The below demos shows the part of the work pipeline. However, we have not wrapped the work flow pipeline into the function yet. In the R package version, the below work flow will be present in the several independent R function with pre-defined parameters.</p>
<div class="section level3">
<h3 id="demo-1-search-cancer-specific-datasets-with-keyword-breast-in-homo-sapiens">4.1 Demo 1: Search Cancer-Specific Datasets with Keyword ‘breast’ in Homo Sapiens<a class="anchor" aria-label="anchor" href="#demo-1-search-cancer-specific-datasets-with-keyword-breast-in-homo-sapiens"></a>
</h3>
<p>To enable the searching function in GEObridge, we need to save the GSE table into current workspace. Here, we just extract top 2,000 datasets to save time.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">search_res</span> <span class="op">=</span> <span class="fu"><a href="../reference/keyword_search.html">keyword_search</a></span><span class="op">(</span>key1<span class="op">=</span><span class="st">'breast'</span>, key2<span class="op">=</span><span class="st">'cancer'</span>, limitN<span class="op">=</span><span class="fl">2000</span>, verbose<span class="op">=</span><span class="cn">TRUE</span>, db<span class="op">=</span><span class="va">connection</span>, subTable<span class="op">=</span><span class="st">'gse'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Total dataset size:"</span></span>
<span><span class="co">## [1] 2000   18</span></span>
<span><span class="co">## [1] "First keyword searched dataset size:"</span></span>
<span><span class="co">## [1] 156</span></span>
<span><span class="co">## [1] "Second keyword searched dataset size:"</span></span>
<span><span class="co">## [1] 462</span></span></code></pre>
<p>The user then can download these dataset using the access GSE ID in the GEO database and perform the different bioinformatics analysis.</p>
</div>
<div class="section level3">
<h3 id="demo-2-summarize-and-compare-the-number-of-dataset-using-different-platform">4.2 Demo 2: Summarize and Compare the Number of Dataset using Different Platform<a class="anchor" aria-label="anchor" href="#demo-2-summarize-and-compare-the-number-of-dataset-using-different-platform"></a>
</h3>
<p>In the downloaded meta data, we can summary all the platfrom collected in the GEO database and print top frequent platform in order. In the first 2,000 dataset, there are total xxx different platforms collected in the database. We only print the top 10 popular platforms here.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top10</span> <span class="op">=</span> <span class="fu"><a href="../reference/print_top_N_platform.html">print_top_N_platform</a></span><span class="op">(</span>topN<span class="op">=</span><span class="fl">10</span>, verbose<span class="op">=</span><span class="cn">TRUE</span>, db<span class="op">=</span><span class="va">connection</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##                                                                    Expression profiling by array </span></span>
<span><span class="co">##                                                                                             1730 </span></span>
<span><span class="co">##                                                                Non-coding RNA profiling by array </span></span>
<span><span class="co">##                                                                                               49 </span></span>
<span><span class="co">##                                               Expression profiling by high throughput sequencing </span></span>
<span><span class="co">##                                                                                               40 </span></span>
<span><span class="co">##                                                          Genome variation profiling by SNP array </span></span>
<span><span class="co">##                                                                                               19 </span></span>
<span><span class="co">## Expression profiling by array;\tGenome binding/occupancy profiling by high throughput sequencing </span></span>
<span><span class="co">##                                                                                               18 </span></span>
<span><span class="co">##                                Expression profiling by array;\tNon-coding RNA profiling by array </span></span>
<span><span class="co">##                                                                                               18 </span></span>
<span><span class="co">##                            Genome variation profiling by SNP array;\tSNP genotyping by SNP array </span></span>
<span><span class="co">##                                                                                               11 </span></span>
<span><span class="co">##               Expression profiling by array;\tExpression profiling by high throughput sequencing </span></span>
<span><span class="co">##                                                                                               10 </span></span>
<span><span class="co">##                                        Genome binding/occupancy profiling by genome tiling array </span></span>
<span><span class="co">##                                                                                               10 </span></span>
<span><span class="co">##                     Expression profiling by array;\tMethylation profiling by genome tiling array </span></span>
<span><span class="co">##                                                                                                6</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="demo-3-visualize-the-result-of-datasets">4.3 Demo 3: Visualize the Result of Datasets<a class="anchor" aria-label="anchor" href="#demo-3-visualize-the-result-of-datasets"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/barplot_top_N_platform.html">barplot_top_N_platform</a></span><span class="op">(</span><span class="va">top10</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignettes_files/figure-html/demo3.1-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="demo-4-interactive-barplot-with-shiny-gadget">4.4 Demo 4: Interactive Barplot with Shiny Gadget<a class="anchor" aria-label="anchor" href="#demo-4-interactive-barplot-with-shiny-gadget"></a>
</h3>
<p>One disadvantage of the above function is that the figure will always show all top frequent platforms. However, the user want to only plot only several interested platforms in the barplot.</p>
<p>Due to the static document cannot show the unpublished interactive Shiny Gadget, we will demostrate this function in the video. The user can also call this function in the RStudio console. The input parameter is the variable generated by last step. After the function <strong>barplot_interact()</strong> running, the RStudio will pop out to the <em>Viewer</em> panel at the right side. In the Shiny Gadget, the user can select the several platforms provided by the input variable. Once the user click the box in front of the selected platforms, the right barplot will update instantly. This can help the user generate a publish qulity figure with their interested platforms.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Due to the static document cannot show the interactive Shiny Gadget, we will demostrate this function in the video.</span></span>
<span><span class="co"># However, you can run below code in your RStudio to play with it.</span></span>
<span></span>
<span><span class="co"># barplot_interact(top10)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="r-environment">4.5 R environment<a class="anchor" aria-label="anchor" href="#r-environment"></a>
</h3>
<p>During the package development, I searched multiple github issues and respositories. Most professional software development engineers suggest to list the R environment if you ask a question or request the help for debug. The R environment will provide a comprehensive list which also help other people reproduce the result or bug.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.0.2 (2020-06-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Catalina 10.15.7</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] GEObridge_0.1.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] Rcpp_1.0.9          tidyr_1.2.0         shinyWidgets_0.7.5 </span></span>
<span><span class="co">##  [4] rprojroot_2.0.3     digest_0.6.30       utf8_1.2.2         </span></span>
<span><span class="co">##  [7] mime_0.12           R6_2.5.1            RSQLite_2.2.12     </span></span>
<span><span class="co">## [10] evaluate_0.17       highr_0.9           ggplot2_3.3.6      </span></span>
<span><span class="co">## [13] pillar_1.8.0        rlang_1.0.6         rstudioapi_0.14    </span></span>
<span><span class="co">## [16] jquerylib_0.1.4     blob_1.2.3          rmarkdown_2.17     </span></span>
<span><span class="co">## [19] pkgdown_2.0.6       labeling_0.4.2      textshaping_0.3.6  </span></span>
<span><span class="co">## [22] desc_1.4.2          readr_2.1.2         stringr_1.4.1      </span></span>
<span><span class="co">## [25] bit_4.0.4           munsell_0.5.0       shiny_1.7.3        </span></span>
<span><span class="co">## [28] compiler_4.0.2      httpuv_1.6.6        xfun_0.30          </span></span>
<span><span class="co">## [31] pkgconfig_2.0.3     systemfonts_1.0.4   BiocGenerics_0.36.1</span></span>
<span><span class="co">## [34] htmltools_0.5.3     tidyselect_1.1.2    tibble_3.1.6       </span></span>
<span><span class="co">## [37] GEOquery_2.58.0     fansi_1.0.3         dplyr_1.0.8        </span></span>
<span><span class="co">## [40] tzdb_0.3.0          later_1.3.0         grid_4.0.2         </span></span>
<span><span class="co">## [43] xtable_1.8-4        jsonlite_1.8.3      gtable_0.3.1       </span></span>
<span><span class="co">## [46] lifecycle_1.0.1     DBI_1.1.3           magrittr_2.0.3     </span></span>
<span><span class="co">## [49] scales_1.2.1        cli_3.4.1           stringi_1.7.6      </span></span>
<span><span class="co">## [52] cachem_1.0.6        farver_2.1.0        fs_1.5.2           </span></span>
<span><span class="co">## [55] promises_1.2.0.1    limma_3.46.0        xml2_1.3.3         </span></span>
<span><span class="co">## [58] bslib_0.4.0         ellipsis_0.3.2      ragg_1.2.2         </span></span>
<span><span class="co">## [61] generics_0.1.3      vctrs_0.4.1         tools_4.0.2        </span></span>
<span><span class="co">## [64] bit64_4.0.5         Biobase_2.50.0      glue_1.6.2         </span></span>
<span><span class="co">## [67] purrr_0.3.4         hms_1.1.1           parallel_4.0.2     </span></span>
<span><span class="co">## [70] fastmap_1.1.0       yaml_2.3.5          colorspace_2.0-3   </span></span>
<span><span class="co">## [73] memoise_2.0.1       GEOmetadb_1.52.0    knitr_1.40         </span></span>
<span><span class="co">## [76] sass_0.4.2</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">5. Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>Data-driven research becomes a novel aspect to boost basic research and cancer immunotherapy development. Thus, a professional and easy-to-use tool or package is very important for researchers. In this proposal, we show the background of the novel GEObridge was proposed. In addition, we explain the benefit group with the GEObridge and how the package will help them during their daily work. To evaluate the feasibility and the utility of the proposed R package, we provide three demos (refer to appendix and code) which complete three important tasks. In the demo, we not only provide the code and the generated figure in the example but also show how to use these functions to perform the analysis and enlarge the power of the GEO database. We believe the proposed R package will benefit more and more researchers and boost new research in the different fields of biology, and immunology.</p>
</div>
<div class="section level2">
<h2 id="future-work-and-plans">6. Future Work and Plans<a class="anchor" aria-label="anchor" href="#future-work-and-plans"></a>
</h2>
<div class="section level3">
<h3 id="a-more-powerful-searching-function-in-the-next-version">6.1 A more powerful searching function in the next version<a class="anchor" aria-label="anchor" href="#a-more-powerful-searching-function-in-the-next-version"></a>
</h3>
<p>One limitation of the current version of the package is that the searching function only supports on two keywords searching in the database. However, some disease might require more searching keys such as Diffuse large B-cell lymphoma (DLBCL) or Clear cell renal cell carcinoma (ccRCC). To enable a more powerful searching function with a high flexibility, we proposed a next version searching function. The input parameter could include a list variable which contains multiple searching keywords. A new version of the function is scheduled, under development and will be publish later after the initial version of the package submitted to the The Comprehensive R Archive Network (CRAN).</p>
</div>
<div class="section level3">
<h3 id="more-visualization-functions-under-development">6.2 More visualization functions under development<a class="anchor" aria-label="anchor" href="#more-visualization-functions-under-development"></a>
</h3>
<p>Another limitation of the current version of the package is that we only provide a barplot function for the platform statistics due to the tight time schedule. Even with an interactive tool thanks to RShiny Gadget, the user still want different kinds of the visualization presentation. To do so, we plan to establish more several plot function including pie plot, boxplot, violin plot, ridgeline plot, correlation plot in the following version of the GEObridge package. These plot function will based on the ggplot2 package as well but providing more user friendly options considering our GEObridge scenario. In addition, we will still use the RShiny Gadget function to provide high flexibility plot function to generate publish quality figure.</p>
</div>
<div class="section level3">
<h3 id="a-storage-optimization-method-for-searching-function">6.3 A storage optimization method for searching function<a class="anchor" aria-label="anchor" href="#a-storage-optimization-method-for-searching-function"></a>
</h3>
<p>One of the most difficult problem in most searching like package is the time complexity of the searching function. A golden standard rule is the time-space trade-off in algorithms. To optimize our searching function, we will prepare some frequent searching result in advance and store them as the RData of the package. To find some frequent searching keywords, we will look through the google searching frequent with GEO database criterion. Meanwhile, we plan to provide several prepared datasets in different area such as cancer, pediatric, and plant biology etc. As a result, the user can select their interested area and download pre-prepared datasets to avoid the slow waiting time.</p>
</div>
</div>
<div class="section level2">
<h2 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<p>[1] S. Davis and P. S. Meltzer, “GEOquery: a bridge between the Gene Expression Omnibus (GEO) and BioConductor,” Bioinformatics, vol. 23, no. 14, pp. 1846-1847, 2007.</p>
<p>[2] T. Barrett et al., “NCBI GEO: mining millions of expression profiles—database and tools,” Nucleic acids research, vol. 33, no. suppl_1, pp. D562-D566, 2005.’</p>
<p>[3] T. Barrett et al., “NCBI GEO: mining tens of millions of expression profiles—database and tools update,” Nucleic acids research, vol. 35, no. suppl_1, pp. D760-D765, 2007.’ [4] T. Barrett et al., “NCBI GEO: archive for functional genomics data sets—update,” Nucleic acids research, vol. 41, no. D1, pp. D991-D995, 2012.</p>
<p>[5] Y. Zhu, S. Davis, R. Stephens, P. S. Meltzer, and Y. Chen, “GEOmetadb: powerful alternative search engine for the Gene Expression Omnibus,” Bioinformatics, vol. 24, no. 23, pp. 2798-2800, 2008.</p>
<p>[6] “GEO Browser.” <a href="https://www.ncbi.nlm.nih.gov/geo/browse/" class="external-link uri">https://www.ncbi.nlm.nih.gov/geo/browse/</a>.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Yue Fang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
